src/clj/game/cards/events.clj

```diff
@@ -647,6 +647,102 @@
                                                          (max 0 (min 4 (available-mu state)))
                                                          {:duration :end-of-run})]))}]})
 
+(defcard "Deep Dive"
+ (letfn [(deep-dive-single [cards state side eid shuffle]
+ 	  ;;access the top card from cards - to skip menu when there is only one valid choice
+ 	  ((system-msg state side (str " accesses the only remaining card in R&D."))
+	     (access-card state side eid (first cards))
+	     (if shuffle
+	       (system-msg state side (str " shuffles R&D!"))
+	       (shuffle! state :corp :deck)
+	     )))
+
+	 ;;access one card from a list of cards
+	 (deep-dive-select [cards]
+          {:prompt "Select another card to access"
+	   :waiting-prompt "Runner to access a card"
+	   :not-distinct true
+	   :choices cards
+	   :async true
+	   :effect (req (wait-for (access-card state side target)
+			(system-msg state side (str " shuffles R&D!"))
+			(shuffle! state :corp :deck)
+			(effect-completed state side eid)
+	   	   ))})
+
+	 ;;access exactly one card from a set of cards
+	 ;;then, potentially access a different card from that set
+	 ;;there is probably a better way to do this
```

Comment:
I think you can break down Deep Dive like this:
- Choose and access a card
- Ask for a second pick
- Choose and access again
- Shuffle.

The `wait-for` mechanism has a place for keeping track of information called `async-result`. The awaited function can pass on a value `X` by calling `(effect-completed state side (make-result eid X))`. The next ability then has a binding `async-result` available to access it.

You can use this to create a `choose-and-access` ability that passes on the remaining set-aside cards as an `async-result`. Then your main effect looks something like this:
```
(wait-for (resolve-ability state side choose-and-access card set-aside)
          (resolve-ability state side {:optional
                                       {:prompt "Access another card?"
                                        :yes-ability choose-and-access}}
                           card async-result)
          (shuffle! state side :deck)
          (system-msg state :corp "shuffles R&D"))
```
Here I also use the fact that the 5-fn used all over the engine have "targets" as their fifth parameter, which I pass the `set-aside` cards and the `async-result` returned by the first ability through.